src/plugins/kibana/public/settings/sections/indices/add_data_steps/pattern_review_step/pattern_review_step.js

```diff
@@ -69,17 +84,21 @@ modules.get('apps/settings')
           }
         });
         $scope.$watch('reviewStep.indexPattern.fields', (fields) => {
-          this.dateFields = _.map(_.filter(fields, {type: 'date'}), 'name');
+          this.dateFields = findFieldsByType(fields, 'date');
         }, true);
 
+
+        this.dateFields = findFieldsByType(this.indexPattern.fields, 'date');
+        this.isTimeBased = !_.isEmpty(this.dateFields);
+
         const buildRows = () => {
           this.rows = _.map(this.indexPattern.fields, (field) => {
-            const sampleValue = this.sampleDoc[field.name];
+            const sampleValue = sampleFields[field.name].value;
```

Comment:
Can you look up the type here in the same way that you're looking up the value? It's always bugged me that we're passing the whole sampleFields collection here. Plus it would be more consistent.